Gabriele, there was an error in the java if you didn't notice the error from the pull-request-test:
DeviceComponent.java:140: error: fullpath has private access in NodeInfo
                    System.out.println("at node: " + subtree.getInfo(nidData, Tree.context).fullpath);

--- a/javatraverser/DeviceComponent.java
+++ b/javatraverser/DeviceComponent.java
@@ -137,7 +137,7 @@ public abstract class DeviceComponent extends JPanel
                 } catch(Exception e)
                 {
                     System.out.println("Error writing device data: " + e);
-                    System.out.println("at node: " + subtree.getInfo(nidData, Tree.context).fullpath);
+                    System.out.println("at node: " + subtree.getInfo(nidData, Tree.context).getFullPath());
                     System.out.println(curr_data);
                     throw e;
                 }

